### ISSUE TITLE
Calibrating L2/R2 on Jdm-030

### ISSUE THREAD

**Issue by Naturelmania** [dualshock-tools/ds4-tools#51]
Hello,

First of all thanks for this awesome tool that saves us.

I had a couple of JDM 055 boards and one JDM 001 board which I successfully calibrated their L2/R2 in past.

That is why I went ahead and set the device changes as permanent and calibrated the triggers for my JDM 030.

The values got stuck to 0.85 for left and 0.65 for right triggers. No matter how many times I try to calibrate, it gets stuck as like this.

I try to press ever so lightly to increase these values but no matter what I do, even if I don't press the triggers, it just keeps getting stuck to these values. Any suggestions that I can do to fix it?

**Comment by qwertychouskie** [comment]
Were there any issues with the triggers before calibration?  I've had controllers with trigger issues where it was a physical problem.  In my case, the plastic firm with the carbon traces had some of the traces damaged.  Replacement parts are cheap from e.g. MobileSentrix (I think it was about 1 dollar for the part for my controller).

**Comment by Naturelmania** [comment]
As you have pointed out qwertychouskie, it turned out that it was due to cheap spare parts. I tried swapping them with my other spare parts gamepad and it worked better which in turn I could finish the calibration correctly. It is for jdm 055 therefore didn't fit correctly but it is well enough to point out what is really wrong.
I'll replace the film and rubber contacts anyway.
Interesting part was it was working good before the calibration, after calibration it got stuck with a lover value.
In conclusion
1- Always try temporary calibration first and make it permanent if it succeeds.
2-If the calibration is not working properly, the rubber or contact film might be damaged or low quality.

Thanks a lot.
